components/tabs/tabs.js

```diff
@@ -573,38 +574,37 @@ class Tabs extends LocalizeStaticMixin(ArrowKeysMixin(RtlMixin(LitElement))) {
 		}
 		this._tabInfos = newTabInfos;
 
-		if (!this._initialized && this._tabInfos.length > 0) {
+		if (this._tabInfos.length > 0 && !selectedTabInfo) {
+			this._tabInfos[0].selected = true;
+			selectedTabInfo = this._tabInfos[0];
+		}
 
-			this._initialized = true;
-			if (!selectedTabInfo) {
-				this._tabInfos[0].selected = true;
-				selectedTabInfo = this._tabInfos[0];
-			}
+		await this.updateComplete;
 
-			await this.updateComplete;
+		const animPromises = [];
+		if (!this._initialized && this._tabInfos.length > 0) {
 
+			this._initialized = true;
 			await this._updateTabsContainerWidth(selectedTabInfo);
-			this._updateMeasures();
-			await this._updateScrollPosition(selectedTabInfo);
 
 		} else {
 
-			await this.updateComplete;
-
-			const animPromises = [];
-			this._tabInfos.forEach((info) => {
-				if (info.state === 'adding') animPromises.push(this._animateTabAddition(info));
-				else if (info.state === 'removing') animPromises.push(this._animateTabRemoval(info));
-			});
-
-			Promise.all(animPromises).then(() => {
-				this._updateMeasures();
-				return this._updateScrollPosition(selectedTabInfo);
-			});
+			if (this._tabInfos.length > 1) {
```

Comment:
The tablist is only visible when there's more than one tab, so this conditional ensure that we only animate the addition/removal of the tab when the tablist is visible... otherwise the animation promise will never resolve.